cloudinit/distros/alpine.py

```diff
@@ -202,6 +205,374 @@ def preferred_ntp_clients(self):
+
+            new_line = ":".join(new_list)
+
+            # Replace existing line for user with modified line
+            shadow_contents = shadow_contents.replace(original_line, new_line)
+            LOG.debug("Updating %s for user %s", update_type, name)
+            try:
+                util.write_file(
+                    shadow_file, shadow_contents, omode="w", preserve_mode=True
+                )
+            except IOError as e:
+                util.logexc(LOG, "Failed to update %s file", shadow_file)
+                raise e
+        else:
+            util.logexc(
+                LOG, "Failed to update %s for user %s", shadow_file, name
+            )
+
+    def lock_passwd(self, name):
+        """
+        Lock the password of a user, i.e., disable password logins
+        """
+
+        # Check whether Shadow's or Busybox's version of 'passwd'
+        (_out, err) = subp.subp(["passwd", "--help"])
+        if err != "":
+            first_line = err.splitlines().pop(0)
+            if re.search(r"^BusyBox.*", first_line):
+                # Busybox's 'passwd'
+                use_shadow = False
+            else:
+                # Not Shadow's or Busybox's 'passwd', assume
+                # it functions the same as Shadow's 'passwd'
+                use_shadow = True
+        else:
+            # Shadow's 'passwd --help' does not output anything to stderr
+            use_shadow = True
+
+        # If Shadow's 'passwd' is available then use the generic
+        # lock_passwd function from __init__.py instead.
+        if use_shadow:
+            return super().lock_passwd(name)
+
+        cmd = ["passwd", "-l", name]
+        # Busybox's 'passwd', unlike Shadow's 'passwd', errors
+        # if password is already locked:
+        #
+        #   "passwd: password for user2 is already locked"
+        #
+        # with exit code 1
+        try:
+            (_out, err) = subp.subp(cmd, rcs=[0, 1])
+            if re.search(r"is already locked", err):
+                return True
+        except Exception as e:
+            util.logexc(LOG, "Failed to disable password for user %s", name)
+            raise e
+
+    def expire_passwd(self, user):
+        # Check whether Shadow's or Busybox's version of 'passwd'
+        (_out, err) = subp.subp(["passwd", "--help"])
+        if err != "":
+            first_line = err.splitlines().pop(0)
+            if re.search(r"^BusyBox.*", first_line):
+                # Busybox's 'passwd'
+                use_shadow = False
+            else:
+                # Not Shadow's or Busybox's 'passwd', assume
+                # it functions the same as Shadow's 'passwd'
+                use_shadow = True
+        else:
+            # Shadow's 'passwd --help' does not output anything to stderr
+            use_shadow = True
+
+        # If Shadow's 'passwd' is available then use the generic
+        # expire_passwd function from __init__.py instead.
+        if use_shadow:
```

Comment:
I think we can replace this and the entire if block above it with:
```python
if not err.startswith("BusyBox")
```

Reply:
Done.